### PR TITLE
Add support for lint

### PR DESCRIPTION
**Pull request by mrunalp**
This gives us a starting point to fix the lint errors.

@nalind @runcom @rhatdan PTAL

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by runcom** [comment]
:+1:

**Comment by nalind** [comment]
Is `.tool/lint` generated, or missing?

**Comment by mrunalp** [comment]
@nalind My bad. I had missed adding that file. Added now :)

File: Makefile [hunk 1]
```diff
@@ -50,6 +50,35 @@ test-unit: build ## run the unit tests using VMs
...
+else
+	git-validation -v -run DCO,short-subject -range $(EPOCH_TEST_COMMIT)..HEAD
+endif
+
+.PHONY: install.tools
+
+install.tools: .install.gitvalidation .install.gometalinter .install.md2man
+
+.install.gitvalidation:
+	GOPATH=${SYSTEM_GOPATH} go get github.com/vbatts/git-validation
```
Comment (nalind) [inline Makefile]: We're not currently setting `$SYSTEM_GOPATH`.  Can you add that?

File: .tool/lint [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+for d in $(find . -type d -not -iwholename '*.git*' -a -not -iname '.tool' -a -not -iwholename '*vendor*'); do
+	gometalinter \
+		--exclude='error return value not checked.*(Close|Log|Print).*\(errcheck\)$' \
+		--exclude='.*_test\.go:.*error return value not checked.*\(errcheck\)$' \
+		--exclude='duplicate of.*_test.go.*\(dupl\)$' \
+		--exclude='cmd\/client\/.*\.go.*\(dupl\)$' \
```
Comment (nalind) [inline .tool/lint]: This path probably needs to be changed.

Comment (mrunalp) [inline .tool/lint]: Removed as this doesn't apply to this repo. Thanks!

**Comment by nalind** [comment]
LGTM.

**Pull request merged**